### PR TITLE
feat: Add support for Python builtin sum

### PR DESCRIPTION
**Pull request by navasvarela**
## Changes

Adds support for Python builtin `sum` operation by allowing integers in `__radd__` for numeric values. 

## Merge requirement checklist

* [x] [CONTRIBUTING](https://github.com/NillionNetwork/nada-dsl/blob/main/CONTRIBUTING.md) guidelines followed
* [x] Unit tests added/updated (if applicable)
* [ ] Backwards compatability analysis completed (if applicable). "Will this change require recompilation and upload of user programs?"

### CODE REVIEW

File: nada_dsl/compile_test.py [hunk 1]
```diff
@@ -36,6 +36,14 @@ def test_compile_nada_fn_simple():
     assert len(mir_functions) == 1
 
 
+def test_compile_sum_integers():
+    mir_str = compile_script(f"{get_test_programs_folder()}/nada_fn_simple.py").mir
```
Comment (lumasepa) [inline nada_dsl/compile_test.py:40]:
```suggestion
    mir_str = compile_script(f"{get_test_programs_folder()}/sum_integers.py").mir
```

**Review by lumasepa — CHANGES_REQUESTED** [review]

File: nada_dsl/compile_test.py (hunk 1 above)
Comment (navasvarela) [reply nada_dsl/compile_test.py:40]: ouch! good catch! Thank you!

File: nada_dsl/nada_types/scalar_types.py [hunk 2]
```diff
@@ -161,6 +161,19 @@ def __ge__(self, other):
...
         )
 
+    def __radd__(self, other):
+        """This adds support for builtin `sum` operation for numeric types."""
+        if isinstance(other, int):
+            other_type = NumericType(
+                Literal(value=other, source_ref=SourceRef.back_frame()),
+                base_type=self.base_type,
+                mode=Mode.CONSTANT,
+            )
```
Comment (pablojhl) [inline nada_dsl/nada_types/scalar_types.py:171]: Shouldn't you use `new_scalar_type(mode, base_type)(Literal(value=other, source_ref=SourceRef.back_frame()))` to create the specific instance of Integer or UnsingedInteger?

Comment (pablojhl) [reply nada_dsl/nada_types/scalar_types.py:171]: In addition, I would say that if you use the factory I mentioned above, the following line (`other_type.value = other`) is not necessary

Comment (navasvarela) [reply nada_dsl/nada_types/scalar_types.py:171]: I wasn't familiar with this constructor, thank you!

Comment (navasvarela) [reply nada_dsl/nada_types/scalar_types.py:171]: Actually your proposal is not correct, it leads to wrong results. Please reconsider your comment.

Comment (pablojhl) [reply nada_dsl/nada_types/scalar_types.py:171]: I'm sorry, the correct replacement should be `new_scalar_type(Mode.CONSTANT, base_type)(Literal(value=other, source_ref=SourceRef.back_frame())`

**Review by Jmgr — APPROVED** [review]

**Review by lumasepa — APPROVED** [review]

**Review by pablojhl — APPROVED** [review]

**Pull request merged**